sklearn/tree/_criterion.pxd

```diff
@@ -52,10 +44,27 @@ cdef class Criterion:
                                 double* impurity_right) nogil
     cdef void node_value(self, double* dest) nogil
     cdef double impurity_improvement(self, double impurity_parent,
-                                     double impurity_left,
-                                     double impurity_right) nogil
+                                    double impurity_left,
+                                    double impurity_right) nogil
     cdef double proxy_impurity_improvement(self) nogil
 
+cdef class Criterion(BaseCriterion):
+    """Abstract interface for supervised impurity criteria."""
+
+    # Internal structures
+    cdef const DOUBLE_t[:, ::1] y        # Values of y
```

Comment:
I think those comments aren't really given extra informations about this instance attribute.
```suggestion
    cdef const DOUBLE_t[:, ::1] y
```